src/node_dlopen.cc

```diff
@@ -0,0 +1,118 @@
+#include "node.h"
+#include "node_buffer.h"
+#include "v8.h"
+#include "env.h"
+#include "env-inl.h"
+
+namespace node {
+namespace dlopen {
+
+using v8::Boolean;
+using v8::Context;
+using v8::FunctionCallbackInfo;
+using v8::Handle;
+using v8::Integer;
+using v8::Local;
+using v8::Number;
+using v8::Object;
+using v8::String;
+using v8::Value;
+using v8::Uint32;
+
+
+static void Dlopen(const FunctionCallbackInfo<Value>& args) {
+  Environment* env = Environment::GetCurrent(args);
+
+  const char* filename;
+  if (args[0]->IsNull()) {
+    filename = nullptr;
+  } else if (args[0]->IsString()) {
+    node::Utf8Value name(env->isolate(), args[0]);
+    filename = *name;
+  } else {
+    return env->ThrowTypeError(
+        "expected a string filename or null as first argument"
+    );
+  }
+
+  if (!Buffer::HasInstance(args[1]))
+    return env->ThrowTypeError("expected a Buffer instance as second argument");
+
+  Local<Object> buf = args[1].As<Object>();
+  uv_lib_t* lib = reinterpret_cast<uv_lib_t*>(Buffer::Data(buf));
```

Comment:
Here is where I'd, instead of using Buffer, would create a `Local<External>` to which you pass the memory pointer. Then it would need to be made a `Persistent<External>`, and then made weak so you can dispose of the memory when it's cleaned up by GC. That would be a reliably future proof way to make this call.

Because once Buffer switches to using `Uint8Array` you'll have to use `Uint8Array::New()`. If you'd like V8 to manage the lifetime of the pointer you can then pass in `kInternalized`. In which case the data will need to be accessed via `ArrayBuffer::Externalize()` (an API that doesn't exist until v4.3). Of you can decide to manage the lifetime of the object. In that case it will be created externalized. Then you'll still need to create a `Persistent<Uint8Array>` and make it weak so the pointer can be cleaned up.

Problem is that the former `ArrayBuffer` approach isn't viable until v4.3 is released. Not sure of any caveats with the later. But either way I'd prefer to not have to rewrite this when that time comes.